### PR TITLE
Parsing the most basic screwdriver.yaml parsing (supports just steps)

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This parses the most basic screwdriver.yaml:

``` yaml
jobs:
  main:
    steps:
      foo: bar
```

and produces data for the https://www.npmjs.com/package/screwdriver-launcher-execute

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            .default('main')
+    });
+    const defaultOptions = {
+        abortEarly: false
+    };
+    let globalConfig = {};
+
+    Async.waterfall([
+        Async.apply(Joi.validate, config, paramSchema, defaultOptions),
+        (parsedConfig, next) => {
```
Comment (FenrirUnbound) [inline index.js:89]: Regarding your earlier concern about it being unreadable: currently the steps are:
1. Validate passed-in config
2. Load global config and validate passed-in config against global
3. Reach for job specifics to execute

Would it be any improvement in readability if the steps were:
1. Read configs
2. Validate configs
3. Reach for job specifics to execute

**Comment by FenrirUnbound** [comment]
👍 

Looks good!

tumblr_o2fzj02via1rr4vd5o2_r1_500

**Comment by nkatzman** [comment]
Looks fine 👍

File: package.json [hunk 2]
```diff
@@ -1,12 +1,15 @@
 {
   "name": "screwdriver-config-parser",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
Comment (d2lam) [inline package.json:3]: Should we bump major version?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            globalConfig = Object.assign({}, parsedConfig);
+
+            try {
+                globalConfig.yaml = Yaml.safeLoad(parsedConfig.yaml);
+            } catch (parseError) {
+                return next(parseError);
+            }
+
+            const configSchema = makeConfigSchema(parsedConfig.jobName);
```
Comment (d2lam) [inline index.js:98]: So right now the yaml can only contain 1 other job beside `main` right?

Comment (stjohnjohnson) [inline index.js:98]: That's only required jobs.

File: package.json (hunk 2 above)
Comment (stjohnjohnson) [inline package.json:3]: Not yet, it's not major enough.

**Comment by d2lam** [comment]
👍

**Pull request merged**